mv/MBS_MobileDirPad.js

```diff
@@ -287,85 +725,118 @@ MBS.MobileDirPad = {};
 	//
 	// The base scene class for all other scenes
 
-	var Scene_Base_start = Scene_Base.prototype.start;
-	var Scene_Base_update = Scene_Base.prototype.update;
-
 	Scene_Base.prototype.isMobileDevice = function() {
-		return Utils.isMobileDevice() || $.Param.pcDebug;
+		return Utils.isMobileDevice() || MBS.MobileDirPad.Param.pcDebug;
 	};
 
 	Scene_Base.dirpad = true;
 
+	MBS.MobileDirPad.Scene_Base_start = Scene_Base.prototype.start;
 	Scene_Base.prototype.start = function() {
-	    Scene_Base_start.apply(this, arguments);
+	    MBS.MobileDirPad.Scene_Base_start.apply(this, arguments);
 	    Scene_Base.dirpad = Scene_Base.dirpad && this.isMobileDevice();
 
-	    if (!$.Param.onlyMap || this instanceof Scene_Map) {
+	    if (!MBS.MobileDirPad.Param.onlyInMap || this instanceof Scene_Map) {
 		    this.createDirPad();
 		    this.createActionButtons();
-		    $.enable(Scene_Base.dirpad);
+		    MBS.MobileDirPad.enable(Scene_Base.dirpad);
 		}
 	};
 
+	MBS.MobileDirPad.Scene_Base_update = Scene_Base.prototype.update;
 	Scene_Base.prototype.update = function() {
-		Scene_Base_update.apply(this, arguments);
-		if (this.isMobileDevice() && this._dirPad != undefined && this._aButton != undefined && this._cButton != undefined)
-			this._dirPad.visible = this._aButton.visible = this._cButton.visible = Scene_Base.dirpad;
+		MBS.MobileDirPad.Scene_Base_update.apply(this, arguments);
+		if (this.isMobileDevice() && this._dirPad != undefined && this._aButton != undefined && this._cButton != undefined && this._shiftButton != undefined && this._mButton != undefined && this._qButton != undefined && this._wButton != undefined)
```

Comment:
This is also bad. I think you'd have to have the buttons in an array and apply some functional programming to check all them at once (see [`Array.some`](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array/some) and [`Array.every`](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array/every)).

The original plugin was bad at that too, but adding more buttons really accentuates this issue.